src/SWP/Bundle/MultiTenancyBundle/Factory/OrganizationFactory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher MultiTenancy Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\MultiTenancyBundle\Factory;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\ODM\PHPCR\HierarchyInterface;
+use SWP\Component\MultiTenancy\Factory\OrganizationFactoryInterface;
+
+class OrganizationFactory implements OrganizationFactoryInterface
+{
+    /**
+     * @var OrganizationFactoryInterface
+     */
+    protected $decoratedFactory;
+
+    /**
+     * @var ObjectManager
+     */
+    protected $documentManager;
+
+    /**
+     * @var string|null
+     */
+    protected $rootPath;
+
+    /**
+     * OrganizationFactory constructor.
+     *
+     * @param OrganizationFactoryInterface $decoratedFactory
+     * @param ObjectManager                $documentManager
+     * @param null                         $rootPath
+     */
+    public function __construct(
+        OrganizationFactoryInterface $decoratedFactory,
+        ObjectManager $documentManager,
+        $rootPath = null
+    ) {
+        $this->decoratedFactory = $decoratedFactory;
+        $this->documentManager = $documentManager;
+        $this->rootPath = $rootPath;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create()
```

Comment:
So it should be possible to create an organization without a code? If not, I think that this method shouldn't be here.